dashboard/server/index.js

```diff
@@ -2,12 +2,35 @@
 const socket = require('socket.io');
 const server = require('./server');
 
-const api = require('./api/index');
+const handlers = require('./api/index');
+const utils = require('../lib/utils');
 
 // setup socket.io server
 const io = socket(server);
 
+// middleware to extract id from the newman run
+io.use(utils.socketAuth);
+
 // event listener for a new connection
-io.on('connection', api.testSocketConnection);
+io.on('connection', (socket) => {
+    const api = handlers(socket);
+
+    if (socket.meta.type === 'newman-run') {
```

Comment:
We can move strings like `newman-run` and `frontend` etc to a constants file.

Reply:
Yes, will do that!. What would be the correct location of this file? In the root itself? Because both reporter and dashboard will use it

Reply:
I think both dashboard and reporter should have separate constants file in their own root dirs. Will make it easy to separate later on.

Reply:
Done the refactoring for the dashboard! I have left the reporter because we aren't sure of the events to listen for and their names until we integrate it with newman